engine/src/main/java/org/hibernate/search/engine/spi/DocumentBuilderIndexedEntity.java

```diff
@@ -635,6 +640,9 @@ else if ( currentFieldValue instanceof Map ) {
 						objectInitializer.initializeMap( (Map) currentFieldValue );
 					}
 				}
+				else if ( member.isArray() && currentFieldValue instanceof Object[] ) {
```

Comment:
It's related, since the commit makes it more likely that arrays will be used, but yeah, it addresses a different issue.

It's just nitpicking anyway: `initializeArray` is a no-op.